src/main/scala/org/broadinstitute/dsde/firecloud/utils/RestJsonClient.scala

```diff
@@ -22,39 +23,49 @@ trait RestJsonClient extends FireCloudRequestBuilding {
   implicit val system: ActorSystem
   implicit val executionContext: ExecutionContext
 
-  def unAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false): Future[HttpResponse] = {
+  def unAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None): Future[HttpResponse] = {
     implicit val userInfo:WithAccessToken = null
-    doRequest(None)(req, compressed, useFireCloudHeader)
+    doRequest(None)(req, compressed, useFireCloudHeader, connector)
   }
-  def userAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false)(implicit userInfo: WithAccessToken): Future[HttpResponse] = doRequest(Option(addCredentials(userInfo.accessToken)))(req, compressed, useFireCloudHeader)
-  def adminAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false): Future[HttpResponse] = doRequest(Option(addAdminCredentials))(req, compressed, useFireCloudHeader)
 
-  private def doRequest(addCreds: Option[RequestTransformer])(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false): Future[HttpResponse] = {
+  def userAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None)(implicit userInfo: WithAccessToken): Future[HttpResponse] =
+    doRequest(Option(addCredentials(userInfo.accessToken)))(req, compressed, useFireCloudHeader, connector)
+
+  def adminAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None): Future[HttpResponse] =
+    doRequest(Option(addAdminCredentials))(req, compressed, useFireCloudHeader, connector)
+
+  private def doRequest(addCreds: Option[RequestTransformer])(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None): Future[HttpResponse] = {
+    val sr = if (connector.isDefined) {
+      sendReceive(connector.get)(executionContext, futureTimeout = 60.seconds)
```

Comment:
here again, this timeout is NOT the timeout for the request itself; it's the timeout for communicating with the transport actor. 60 seconds is spray's default; see https://github.com/spray/spray/blob/master/spray-client/src/main/scala/spray/client/pipelining.scala#L34.

This is a good explanation of timeouts: http://kamon.io/teamblog/2014/11/02/understanding-spray-client-timeout-settings/